apps/demos/Demos/Chat/Customization/ReactJs/App.js

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useCallback } from 'react';
+import Chat from 'devextreme-react/chat';
+import SelectBox from 'devextreme-react/select-box';
+import CheckBox from 'devextreme-react/check-box';
+import {
+  currentUser,
+  messages as initialMessages,
+  dayHeaderFormats as headerFormats,
+  messageTimestampFormats as messageTimestamps,
+  messageTimestampLabel,
+  dayHeaderLabel,
+} from './data.js';
+
+export default function App() {
+  const [messages, setMessages] = useState(initialMessages);
+  const [showAvatar, setShowAvatar] = useState(true);
+  const [showUsername, setShowUsername] = useState(true);
+  const [showDayHeaders, setDayHeaders] = useState(true);
+  const [dayHeaderFormat, setDayHeaderFormat] = useState(headerFormats[0]);
+  const [showMessageTime, setMessageTime] = useState(true);
+  const [messageTimestampFormat, setMessageTimeFormat] = useState(messageTimestamps[0]);
+  const [isDisabled, setDisabled] = useState(false);
+  function onMessageEntered({ message }) {
+    setMessages((prevMessages) => [...prevMessages, message]);
+  }
+  const dayHeaderFormatChanged = useCallback(
```

Comment:
In React, we can use a shorter notation to mutate the state when the checkbox value changes
 
const [showDayHeaders, setDayHeaders] = useState(true);
 
<CheckBox
            text="Day Header"
            value={showDayHeaders}
            onValueChange={setDayHeaders}
          />
 
It allows you to avoid declaring an additional handler
 const showDayHeaderChanged = useCallback(
    (e) => {
      setDayHeaders(e.value);
    },
    [setDayHeaders],
  );

Reply:
[3f9ad9a](https://github.com/DevExpress/DevExtreme/pull/28482/commits/3f9ad9add7cff2a3304c33b878339d81e15b49de)